### PR TITLE
Boards will now only size to squares for new games and piece creation

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java [hunk 1]
```diff
@@ -35,11 +37,56 @@ public BoardPanel(BoardSize boardSize, SquareConfig squareConfig,
...
+                else{
+                    setMinimumSize(new Dimension(b.height, b.height));
+                    setPreferredSize(new Dimension(b.height, b.height));
+                    setMaximumSize(new Dimension(b.height, b.height));
+                }
+            }
+
+            @Override
+            public void componentMoved(ComponentEvent e) {
+
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java]: Extra newlines here

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java [hunk 2]
```diff
@@ -35,11 +37,56 @@ public BoardPanel(BoardSize boardSize, SquareConfig squareConfig,
         mBoardSize = boardSize;
         mSquareLabels = new ArrayList<>(boardSize.width * boardSize.height);
 
+        addComponentListener(new ComponentListener() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                Rectangle b = e.getComponent().getBounds();
+                if(b.height > b.width){
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java]: How about:

```
int size = Math.min(b.height, b.width);
setMinimumSize(new Dimension(s, s));
setPreferredSize(new Dimension(s, s));
setMaximumSize(new Dimension(s, s));
```

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java [hunk 3]
```diff
@@ -35,11 +37,56 @@ public BoardPanel(BoardSize boardSize, SquareConfig squareConfig,
...
+            }
+        });
+
         GridLayout gridLayout = new GridLayout(mBoardSize.width + 2, mBoardSize.height + 2);
         setLayout(gridLayout);
         createGrid(gridLayout);
     }
 
+    public void updateDimensions(int width, int height){
+        if(height > width){
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java]: Same comment as above. In fact, why not use the same code in both places? No duplication

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java [hunk 4]
```diff
@@ -76,11 +81,34 @@ public GamePanel(@NotNull GlassPane glassPane, @NotNull Game game) {
...
+        setLayout(new BoxLayout(this, BoxLayout.LINE_AXIS));
 
         JPanel boardPanels = new JPanel();
         boardPanels.setOpaque(false);
-        boardPanels.setLayout(new GridBagLayout());
+        boardPanels.setLayout(new BoxLayout(boardPanels, BoxLayout.LINE_AXIS));
+        boardPanels.addComponentListener(new ComponentListener() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                mGameBoards[0].updateDimensions(e.getComponent().getWidth(), e.getComponent().getHeight());
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java]: We need to do this for all boards, right?
`Stream.of(mGameBoards).forEach(...)`

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java [hunk 5]
```diff
@@ -76,11 +81,34 @@ public GamePanel(@NotNull GlassPane glassPane, @NotNull Game game) {
...
+                mGameBoards[0].revalidate();
+                mGameBoards[0].repaint();
+            }
+
+            @Override
+            public void componentMoved(ComponentEvent e) {
+
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java]: Extra newlines

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java [hunk 6]
```diff
@@ -178,44 +193,45 @@ private void initComponents() {
...
+        boardPanels.setOpaque(false);
+        boardPanels.setLayout(new BoxLayout(boardPanels, BoxLayout.LINE_AXIS));
+        boardPanels.add(Box.createHorizontalGlue());
+        boardPanels.add(mBoardPanel);
+        boardPanels.add(Box.createHorizontalGlue());
+        boardPanels.addComponentListener(new ComponentListener() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                mBoardPanel.updateDimensions(e.getComponent().getWidth(), e.getComponent().getHeight());
+                mBoardPanel.revalidate();
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java]: Can `BoardPanel#updateDimensions()` just call `revalidate()` and `repaint()` internally, so we don't have to remember to call it everyplace else?

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java [hunk 7]
```diff
@@ -178,44 +193,45 @@ private void initComponents() {
...
 
-        gbc.gridx = 0;
-        gbc.weightx = 1.0;
-        gbc.weighty = 1.0;
-        gbc.fill = GridBagConstraints.BOTH;
-        boardAndSave.add(mBoardPanel, gbc);
+            }
+
+            @Override
+            public void componentShown(ComponentEvent e) {
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java]: extra newlines

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java (hunk 6 above)
Comment (ghost) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java]: No. It has to be called in the panel that the event is happening on.

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java [hunk 8]
```diff
@@ -35,11 +37,48 @@ public BoardPanel(BoardSize boardSize, SquareConfig squareConfig,
         mBoardSize = boardSize;
         mSquareLabels = new ArrayList<>(boardSize.width * boardSize.height);
 
+        addComponentListener(new ComponentListener() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                int size = Math.min(e.getComponent().getHeight(), e.getComponent().getWidth());
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java]: Why do we need both this and `updateDimensions()`?

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java [hunk 9]
```diff
@@ -35,11 +37,48 @@ public BoardPanel(BoardSize boardSize, SquareConfig squareConfig,
...
+            }
+        });
+
         GridLayout gridLayout = new GridLayout(mBoardSize.width + 2, mBoardSize.height + 2);
         setLayout(gridLayout);
         createGrid(gridLayout);
     }
 
+    public void updateDimensions(int width, int height) {
+        if (height > width) {
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/BoardPanel.java]: Why not do the same `size` trick here?

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java [hunk 10]
```diff
@@ -76,13 +81,39 @@ public GamePanel(@NotNull GlassPane glassPane, @NotNull Game game) {
...
 
         JPanel boardPanels = new JPanel();
         boardPanels.setOpaque(false);
-        boardPanels.setLayout(new GridBagLayout());
+        boardPanels.setLayout(new BoxLayout(boardPanels, BoxLayout.LINE_AXIS));
+        boardPanels.addComponentListener(new ComponentListener() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                IntStream.range(0, boards.length).forEach(boardIndex -> {
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java]: Is there a reason you didn't use the one I suggested?
`Streams.of(mGameBoards).forEach(board -> {...});`

**Pull request merged**